src/backend.c

```diff
@@ -5160,59 +5044,74 @@ int run_opencl_kernel_utf8toutf16le (hashcat_ctx_t *hashcat_ctx, hc_device_param
 
   if (hc_clSetKernelArg (hashcat_ctx, kernel, 1, sizeof (cl_ulong), device_param->kernel_params_utf8toutf16le[1]) == -1) return -1;
 
-  if (hc_clEnqueueNDRangeKernel (hashcat_ctx, device_param->opencl_command_queue, kernel, 1, NULL, global_work_size, local_work_size, 0, NULL, NULL) == -1) return -1;
+  return hc_clEnqueueNDRangeKernel (hashcat_ctx, device_param->opencl_command_queue, kernel, 1, NULL, global_work_size, local_work_size, 0, NULL, NULL);
+}
 
-  if (hc_clFlush (hashcat_ctx, device_param->opencl_command_queue) == -1) return -1;
+int run_opencl_kernel_memset (hashcat_ctx_t *hashcat_ctx, hc_device_param_t *device_param, cl_mem buf, const u64 offset, const u8 value, const u64 size)
+{
+  const backend_ctx_t *backend_ctx = hashcat_ctx->backend_ctx;
+  const OCL_PTR       *ocl         = backend_ctx->ocl;
 
-  if (hc_clFinish (hashcat_ctx, device_param->opencl_command_queue) == -1) return -1;
+  int rc;
 
-  return 0;
+  /* workaround if missing clEnqueueFillBuffer() */
+  if (ocl->clEnqueueFillBuffer == NULL)
+  {
+    const u64 len = offset + size;
```

Comment:
Why offset?

Reply:
This comes from old implementation, which used this. It may have been a bug or workaround for some buggy OpenCL implementation calculating offset to ptr. It should be sufficient to have buffer length equal to size as that is what the specs says.

Reply:
I removed the offset from buffer sizing, fdbfae9